yellowbrick/features/projection.py

```diff
@@ -0,0 +1,303 @@
+            self.ax
+            return
+        # Create the new axes for the colorbar
+        divider = make_axes_locatable(self.ax)
+        self._cax = divider.append_axes("right", size="5%", pad=0.3)
+        self._cax.set_yticks([])
+        self._cax.set_xticks([])
+
+    def fit(self, X, y=None, **kwargs):
+        """
+        Fits the transformer on X. Also identifies the classes, features and
+        colors by calling the super class.
+        
+        Parameters
+        ----------
+        X : ndarray or DataFrame of shape n x m
+            A matrix or data frame of n instances with m features where m>2.
+
+        y : array-like of shape (n,), optional
+            A vector or series with target values for each instance in X. This
+            vector is used to determine the color of the points in X.
+
+        Returns
+        -------
+        self : instance
+            Returns the instance of the transformer/visualizer.
+        """
+        super(ProjectionVisualizer, self).fit(X, y, **kwargs)
+        self.transformer.fit(X)
+        return self
+
+    def transform(self, X, y=None, **kwargs):
+        """
+        Returns the transformed data points from the transformers. It also calls
+        draw which draws a scatter plot with transformed data.
+        
+        Parameters
+        ----------
+        X : ndarray or DataFrame of shape n x m
+            A matrix or data frame of n instances with m features. m must be
+            same as that used in ``fit()`` method.
+
+        y : array-like of shape (n,), optional
+            A vector or series with target values for each instance in X. This
+            vector is used to determine scatter arguments.
+
+        Returns
+        -------
+        Xprime : array-like of shape (n, 2)
+            Returns the 2-dimensional embedding of the instances.
+        """
+        try:
+            Xp = self.transformer.transform(X)
+        except AttributeError as e:
+            raise AttributeError(str(e) + " try using fit_transform instead.")
+        self.draw(Xp, y, **kwargs)
+        return Xp
+    
+    def fit_transform(self, X, y=None, **kwargs):
+        """
+        Fit to data, then transform it.
+
+        Fits transformer to X and y with optional parameters fit_params
+        and returns a transformed version of X.
+
+        Parameters
+        ----------
+        X : ndarray or DataFrame of shape n x m
+            A matrix or data frame of n instances with m features where m>2.
+
+        y : array-like of shape (n,), optional
+            A vector or series with target values for each instance in X. This
+            vector is used to determine the color of the points in X.
+
+        Returns
+        -------
+        Xprime : array-like of shape (n, 2)
+            Returns the 2-dimensional embedding of the instances.
+
+        """
+        return self.fit(X, y, **kwargs).transform(X, y, **kwargs)
+    
+    def draw(self, X, y=None):
```

Comment:
Could we please change the name of `X` to `Xp` to reflect that the transformed data is passed to draw?

Reply:
Sure!